File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/consumer/OrderEventsHandler.scala
package com.example.events.consumer

import cats.effect.IO
import com.example.events.OrderCancelled
import com.example.events.OrderEvents
import com.example.events.OrderPlaced
import com.example.events.OrderUpdated
import com.example.events.header.StandardHeaders
import java.lang.IllegalStateException

/** Handler interface for order-events topic events */
trait OrderEventsHandler {
  /** Handle a OrderCancelled event */
  def handleOrderCancelled(
    key: String,
    event: OrderCancelled,
    headers: StandardHeaders
  ): IO[Unit]

  /** Handle a OrderPlaced event */
  def handleOrderPlaced(
    key: String,
    event: OrderPlaced,
    headers: StandardHeaders
  ): IO[Unit]

  /** Handle a OrderUpdated event */
  def handleOrderUpdated(
    key: String,
    event: OrderUpdated,
    headers: StandardHeaders
  ): IO[Unit]

  /** Handle unknown event types (default throws exception) */
  def handleUnknown(
    key: String,
    event: OrderEvents,
    headers: StandardHeaders
  ): IO[Unit] = {
    throw new IllegalStateException("Unknown event type: " + event.getClass)
  }
}

File: testers/avro/java-vanilla/generated-and-checked-in/com/example/events/consumer/OrderEventsHandler.java
package com.example.events.consumer;

import com.example.events.OrderCancelled;
import com.example.events.OrderEvents;
import com.example.events.OrderPlaced;
import com.example.events.OrderUpdated;
import com.example.events.header.StandardHeaders;

/** Handler interface for order-events topic events */
public interface OrderEventsHandler {
  /** Handle a OrderCancelled event */
  void handleOrderCancelled(String key, OrderCancelled event, StandardHeaders headers);

  /** Handle a OrderPlaced event */
  void handleOrderPlaced(String key, OrderPlaced event, StandardHeaders headers);

  /** Handle a OrderUpdated event */
  void handleOrderUpdated(String key, OrderUpdated event, StandardHeaders headers);

  /** Handle unknown event types (default throws exception) */
  default void handleUnknown(String key, OrderEvents event, StandardHeaders headers) {
    throw new IllegalStateException("Unknown event type: " + event.getClass());
  }
}
